native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/MetadataCopyMojo.java

```diff
@@ -0,0 +1,45 @@
+package org.graalvm.buildtools.maven;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.graalvm.buildtools.maven.config.MetadataCopyConfiguration;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Mojo(name = "metadata-copy", defaultPhase = LifecyclePhase.PREPARE_PACKAGE)
+public class MetadataCopyMojo extends AbstractMojo {
+
+    @Parameter(alias = "metadataCopy")
+    private MetadataCopyConfiguration metadataCopy;
+
+    @Parameter(defaultValue = "${project}", readonly = true, required = true)
+    private MavenProject project;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        ArrayList<Path> sources = new ArrayList<>(Arrays.asList(Paths.get(project.getBuild().getOutputDirectory()), Paths.get(project.getBuild().getTestOutputDirectory())));
+        ArrayList<Path> destinations = metadataCopy.getOutputDirectories();
+
+        for (Path source: sources) {
+            for (Path destination : destinations) {
+                try {
+                    Files.copy(source, destination);
```

Comment:
This isn't correct. `Files.copy()` copies a single file.

Also, `project.getBuild().getOutputDirectory()` is `target/classes` and `project.getBuild().getTestOutputDirectory()` is `target/test-classes`, but that is not where the agent dumps the configuration files. 

You need to copy **all files** from `target/native/agent-output/main` and `target/native/agent-output/test` into the destination(s).